shoop/admin/modules/contact_groups/views/members/forms.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from django import forms
+from django.contrib import messages
+from django.db.transaction import atomic
+from django.forms.formsets import BaseFormSet, DELETION_FIELD_NAME
+from django.utils.translation import ugettext_lazy as _
+
+from shoop.admin.forms.widgets import ContactChoiceWidget
+from shoop.core.models import Contact
+
+
+class MembersForm(forms.Form):
+    member = forms.ModelChoiceField(
+        queryset=Contact.objects.all(),
+        widget=ContactChoiceWidget(empty_text=""),
+        label=_('member')
+    )
+
+
+class MembersFormSet(BaseFormSet):
+    def __init__(self, **kwargs):
+        kwargs.pop("empty_permitted", None)
+        self.request = kwargs.pop("request", None)
+        self.contact_group = kwargs.pop("contact_group")
+        kwargs["initial"] = [
+            {"member": member}
+            for member
+            in self.contact_group.get_members()
+            ]
+        super(MembersFormSet, self).__init__(**kwargs)
+
+    def _construct_form(self, i, **kwargs):
+        form = super(MembersFormSet, self)._construct_form(i, **kwargs)
+        form.fields[DELETION_FIELD_NAME].label = _("Remove")
+        return form
+
+    def save(self):
+        contact_group = self.contact_group
+        current_members = set(contact_group.get_members())
+        selected_members, removed_members = self.get_selected_and_removed()
+
+        with atomic():
+            members_to_add = selected_members - current_members
+            members_to_remove = current_members & removed_members
+            for member in members_to_remove:
+                contact_group.members.remove(member)
+            for member in members_to_add:
+                contact_group.members.add(member)
+
+        message_parts = []
+        if members_to_add:
+            message_parts.append(_("New: %d") % len(members_to_add))
```

Comment:
Better message would be "%(count)s new members added" or something. This new is not descriptive enough IMO.